app/views/products/_form.html.erb

```diff
@@ -0,0 +1,8 @@
+<%= form_for [ @vendor, @product ] do |f| %>
+  <%= f.label :name, "Name:" %>
+  <%= f.text_field :name, required: :true %>
+
+  <%= f.hidden_field :vendor_id, value: params[:vendor_id] %>
```

Comment:
I commented out the hidden field and edited the Products#create method to be this:

def create
    product_params_with_vendor = product_params
    product_params_with_vendor[:vendor_id] = @vendor[:id]
    product = Product.create(product_params_with_vendor)
    redirect_to vendor_product_path(@vendor, product)
  end

And now it's assigning the vendor id to the new product being created. Is this the type of approach you're suggesting? It feels really clunky, so I feel like I'm not understanding something still... thanks for your patience!